src/Mod/CAM/Path/Base/Generator/helix.py

```diff
@@ -115,26 +126,30 @@ def generate(
     if startPoint.z < endPoint.z:
         raise ValueError("start point is below end point")
 
+    tool_radius = tool_diameter / 2
     if hole_radius <= tool_diameter:
-        Path.Log.debug("(single helix mode)\n")
-        radii = [hole_radius - tool_diameter / 2]
-        if radii[0] <= 0:
+        Path.Log.debug("(single helix path)\n")
+        radii = [hole_radius - tool_radius]
+        if radii[0] < 0:
             raise ValueError(
-                "Cannot helix a hole of diameter {0} with a tool of diameter {1}".format(
+                "Cannot creatre a helical path of diameter {0} with a tool of diameter {1}".format(
                     2 * hole_radius, tool_diameter
                 )
             )
-        outer_radius = hole_radius
-
-    else:  # inner_radius > 0:
-        Path.Log.debug("(annulus mode / full hole)\n")
-        outer_radius = hole_radius - tool_diameter / 2
-        step_radius = inner_radius + tool_diameter / 2
-        if abs((outer_radius - step_radius) / step_over_distance) < 1e-5:
-            radii = [(outer_radius + inner_radius) / 2]
+    else:  # inner_radius >= 0:
```

Comment:
Just a note from the other, now resolved, comment:
Comment on [`else`](https://github.com/FreeCAD/FreeCAD/pull/20537/files#diff-5735e1231038eb2ebfffeb80742a7dddda7c809600e72a1ea54ed044e8530612R139) (`# inner_radius >= 0:`) doesn't match the [`if`](https://github.com/FreeCAD/FreeCAD/pull/20537/files#diff-5735e1231038eb2ebfffeb80742a7dddda7c809600e72a1ea54ed044e8530612R130) (`if hole_radius <= tool_diameter:`).

Reply:
If the other note is resolved why do you relaunch the issue again?  Fortunately you've  discovered how to include the relevant "if" statement using GH. 

As you will note this erroneous comment is already in master.  All I did was add the "=".  So the issue about it not corresponding to the if clause currently still a live issue.  Since it seems to be of significant importance to you maybe you would like to create a PR to remove it entirely.

Reply:
> If the other note is resolved why do you relaunch the issue again? Fortunately you've discovered how to include the relevant "if" statement using GH.

You where unhappy that I didn't include the `if` in the last comment so I replaced it with this new one ("resolved" the old one) as it isn't possible to change scope on github.

> All I did was add the "=".

Before your change it was probably just missed, now you've incorrectly "fixed" it. I don't see it unreasonable to ask you to properly fix the issue, or drop your change if you really don't want to make it correct.

Reply:
I was not unhappy that you didn't include the if, I was unhappy that you had not even bothered reading the code and made false statements about what the code actually contained.  I pointed out that if you had taken the trouble to find and quote the relevant if , your mistake would have been obvious to you and you would not have made false claims. 

If you are going to review code , you need to actually read it first. 

I've now removed the offending comment entirely, so try to move on.